packages/tupaia-web/src/components/RouterButton.tsx

```diff
@@ -14,13 +15,20 @@ interface RouterLinkProps extends Record<string, any> {
 interface ModalButtonProps extends Record<string, any> {
   modal: string;
   children?: ReactNode;
+  removeParams?: string[]; // search params to remove from url when navigating to modal hash route
 }
 
 type RouterButtonProps = RouterLinkProps | ModalButtonProps;
 
-export const RouterButton = ({ to, modal, children, ...props }: RouterButtonProps) => {
+export const RouterButton = ({
+  to,
+  modal,
+  removeParams,
+  children,
+  ...props
+}: RouterButtonProps) => {
   const location = useLocation();
-  const link = modal ? { ...location, hash: modal } : to;
+  const link = modal ? { ...location, hash: modal, search: useRemoveParams(removeParams) } : to;
```

Comment:
I'm not quite sure I understand what is happening here. Do you want the modal links to never have query string params? Why is that?

Reply:
So I made a way to remove specified params, so `removeParams` is an array, and `useRemoveParams` removes any params in that array, otherwise returns as is. There are a few modals, e.g. OneTimeLogin and RequestProjectAccess that have search params as well as hash, and when they close, we want them to remove the associated search params